src/Sentry/Internal/MainSentryEventProcessor.cs

```diff
@@ -96,20 +96,17 @@ public SentryEvent Process(SentryEvent @event)
 
             @event.Platform = Protocol.Constants.Platform;
 
-            if (@event.Sdk != null)
+            // SDK Name/Version might have be already set by an outer package
+            // e.g: ASP.NET Core can set itself as the SDK
+            if (@event.Sdk.Version == null && @event.Sdk.Name == null)
```

Comment:
Are transactions going through this at the end? Doesn't seem like it since this take SentryEvent.

So transactions will always be reported as `sentry.dotnet`?

Reply:
Yeah... The SDK is set separately when creating the transaction right now. Transactions also don't get all the other default metadata either, which is populated by event processors.